packages/shared/src/utils/config/config.ts

```diff
@@ -71,32 +71,32 @@ export const getAppId = () => {
 };
 
 export const getSocketURL = () => {
-    const local_storage_server_url = window.localStorage.getItem('config.server_url');
+    const local_storage_server_url: string | null = window.localStorage.getItem('config.server_url');
     if (local_storage_server_url) return local_storage_server_url;
 
     let active_loginid_from_url;
-    const search = window.location.search;
+    const search: string = window.location.search;
     if (search) {
-        const params = new URLSearchParams(document.location.search.substring(1));
+        const params: URLSearchParams = new URLSearchParams(document.location.search.substring(1));
         active_loginid_from_url = params.get('acct1');
     }
 
-    const loginid = window.localStorage.getItem('active_loginid') || active_loginid_from_url;
-    const is_real = loginid && !/^VRT/.test(loginid);
+    const loginid: string | null | undefined = window.localStorage.getItem('active_loginid') || active_loginid_from_url;
+    const is_real: boolean | '' | null | undefined = loginid && !/^VRT/.test(loginid);
 
```

Comment:
it looks like almost 'any'
and also I believe we shouldn’t use 'undefined' for types at all..
IMHO we should use some default values and have checks for coming props or used values if they are not undefinied